### PR TITLE
keepassxc: 2.4.0 -> 2.4.1

### PR DESCRIPTION
**Pull request by jonafato**
###### Motivation for this change

KeePassXC has a new release. Changelog at
https://github.com/keepassxreboot/keepassxc/releases/tag/2.4.1.

Additionally, some cleanup:

- Alphabetize inputs for better future diffs
- Drop a no-longer-relevant patch
- Fixup removed and missing cmake flags
- Update description to reference correct browser extension

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/keepassx/community.nix [hunk 1]
```diff
@@ -52,17 +54,21 @@ stdenv.mkDerivation rec {
...
     "-DKEEPASSXC_BUILD_TYPE=Release"
     "-DWITH_GUI_TESTS=ON"
     "-DWITH_XC_AUTOTYPE=ON"
+    "-DWITH_XC_UPDATECHECK=OFF"
     "-DWITH_XC_YUBIKEY=ON"
-    "-DWITH_XC_KEESHARE=ON"
   ]
   ++ (optional withKeePassBrowser "-DWITH_XC_BROWSER=ON")
-  ++ (optional withKeePassHTTP "-DWITH_XC_HTTP=ON")
+  ++ (optional withKeePassKeeShare "-DWITH_XC_KEESHARE=ON")
```
Comment (bobvanderlinden) [inline pkgs/applications/misc/keepassx/community.nix]: Consider using `optionals`?

```suggestion
  ++ (optionals withKeePassKeeShare [
    "-DWITH_XC_KEESHARE=ON"
    "-DQUAZIP_INCLUDE_DIR=${quazip}/include/quazip"
    "-DQUAZIP_LIBRARIES=${quazip}/lib/libquazip.so"
  ])
```

File: pkgs/applications/misc/keepassx/community.nix [hunk 2]
```diff
@@ -52,17 +54,21 @@ stdenv.mkDerivation rec {
...
+  ++ (optional withKeePassKeeShare "-DQUAZIP_INCLUDE_DIR=${quazip}/include/quazip")
+  ++ (optional withKeePassKeeShare "-DQUAZIP_LIBRARIES=${quazip}/lib/libquazip.so")
```
Comment (bobvanderlinden) [inline pkgs/applications/misc/keepassx/community.nix]: I have not had time to test this yet. Is this compatible with OSX?

Would it be possible to 'automatically' find this using `pkg-config` by adding `quazip` to the `buildInputs`? It seems keepass does have such functionality: https://github.com/keepassxreboot/keepassxc/blob/b59fd6d06ad0b130c00c8afe8e003999335afd21/cmake/FindQuaZip.cmake

**Review by bobvanderlinden — COMMENTED** [review]
I was able to build and run this on NixOS, but have not tried KeeShare (nor OSX).

**Comment by jonafato** [comment]
> I was able to build and run this on NixOS, but have not tried KeeShare (nor OSX).

KeeShare was already being built, I've just moved it to optional to match what KeePassXC's cmake setup supports. There was a bug in the 2.4.0 release that prevented building without this enabled.

File: pkgs/applications/misc/keepassx/community.nix (hunk 1 above)
Comment (jonafato) [reply pkgs/applications/misc/keepassx/community.nix]: Updated this below with `withKeePassShareSecure` where it should have been originally.

**Comment by bobvanderlinden** [comment]
Sounds good :+1: Works fine on OSX as well btw.

**Review by bobvanderlinden — APPROVED** [review]

File: pkgs/applications/misc/keepassx/community.nix (hunk 2 above)
Comment (jonafato) [reply pkgs/applications/misc/keepassx/community.nix]: Updated to use `pkg-config`. That required an additional line in the patch, but it's probably cleaner overall.

**Pull request merged**